### PR TITLE
[simd] Implement .ctor(T[]) and CopyTo () as intrinsics.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/simd-intrinsics-netcore.c [hunk 1]
```diff
@@ -323,6 +324,66 @@ emit_sys_numerics_vector_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSig
...
+	case SN_CopyTo:
+		if ((fsig->param_count == 1 || fsig->param_count == 2) && (fsig->params [0]->type == MONO_TYPE_SZARRAY)) {
+			MonoInst *array_ins = args [1];
+			MonoInst *index_ins;
+			MonoInst *ldelema_ins;
+			int val_vreg, end_index_reg;
+
+			val_vreg = load_simd_vreg (cfg, cmethod, args [0], NULL);
+
+			/* .ctor (T[]) or .ctor (T[], index) */
```
Comment (lewurm) [inline mono/mini/simd-intrinsics-netcore.c]: copy-pasta

File: mono/mini/simd-intrinsics-netcore.c [hunk 2]
```diff
@@ -597,8 +658,13 @@ mono_emit_simd_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 
 	class_ns = m_class_get_name_space (cmethod->klass);
 	class_name = m_class_get_name (cmethod->klass);
-	if (!strcmp (class_ns, "System.Numerics") && !strcmp (class_name, "Vector"))
-		return emit_sys_numerics_vector (cfg, cmethod, fsig, args);
+	if (!strcmp (class_ns, "System.Numerics") && !strcmp (class_name, "Vector")) {
+		MonoInst *ins = emit_sys_numerics_vector (cfg, cmethod, fsig, args);
+		if (!ins) {
+			//printf ("M: %s %s\n", mono_method_get_full_name (cfg->method), mono_method_get_full_name (cmethod));
```
Comment (lewurm) [inline mono/mini/simd-intrinsics-netcore.c]: debug left over?

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by filipnavara** [comment]
Tests on [Burgers_3 benchmark](https://github.com/dotnet/performance/blob/3941f11e000edd7a404599fc2e401129c25bc4ad/src/benchmarks/micro/coreclr/Burgers/Burgers.cs#L195) show an improvement of over 14 times (before: 2,437.2 ms, after: 170.5 ms). Good job!

**Comment by EgorBo** [comment]
This currently ignores `CopyTo(Span)` doesn't it?

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
Yes, it does.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**